component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordChangeHandler.java

```diff
@@ -42,12 +51,43 @@
 public class PasswordChangeHandler extends AbstractEventHandler implements IdentityConnectorConfig {
     private static final Log log = LogFactory.getLog(PasswordChangeHandler.class);
 
+
     @Override
     public void handleEvent(Event event) throws IdentityEventException {
+
+        String eventName = event.getEventName();
         // Fetching event properties
         String username = (String) event.getEventProperties().get(IdentityEventConstants.EventProperty.USER_NAME);
         UserStoreManager userStoreManager = (UserStoreManager) event.getEventProperties()
                 .get(IdentityEventConstants.EventProperty.USER_STORE_MANAGER);
+
+        //password grant handler - password expiry validation
+        if (PasswordPolicyConstants.PASSWORD_GRANT_POST_AUTHENTICATION_EVENT.equals(eventName)) {
+            String tenantDomain = (String) event.getEventProperties()
+                    .get(IdentityEventConstants.EventProperty.TENANT_DOMAIN);
+            String tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(username);
+            boolean authenticationStatus = (boolean) event.getEventProperties().get(
+                    PasswordPolicyConstants.AUTHENTICATION_STATUS);
+
+            if (authenticationStatus) { // only validate password expiry if user is authenticated
+                if (log.isDebugEnabled()) {
+                    log.debug("Checking password validity of " + username);
+                }
+                try {
+                    if (isPasswordExpired(tenantDomain, tenantAwareUsername, userStoreManager)) {
+                        if (log.isDebugEnabled()) {
+                            log.debug("User: " + username + " password is expired.");
```

Comment:
Haven't we discouraged logging username (due to concerns like GDPR). Shouldn't we log the user identifiers only (UUIDs)?

Reply:
only debug level logs were logged with username